src/Makefile

```diff
@@ -29,6 +29,16 @@ endif
 WARN=-Wall -W -Wno-missing-field-initializers
 OPT=$(OPTIMIZATION)
 
+# Detect if the compiler supports C11 _Atomic
+C11_ATOMIC := $(shell sh -c 'echo "\#include <stdatomic.h>" > foo.c; \
+	$(CC) -std=c11 -c foo.c -o foo.o &> /dev/null; \
+	if [ -f foo.o ]; then echo "yes"; rm foo.o; fi; rm foo.c')
+ifeq ($(C11_ATOMIC),yes)
+	STD+=-std=c11
+else
+	STD+=-std=c99
+endif
```

Comment:
Oh, I missed it. Does it matter?  We has persisted `STD` in `.make-setting`.

Reply:
@yossigo on a second thought, since this is not controlled by a user argument (unlike `noopt`, `32bit` or other targets / arguments), and is actually defined by the toolchain. we'll get issues only when the toolchain is changed, and in that case the persisted setting may not help (build may be bound to fail).

by reading the makefile it seems like it is persisted (modifying `STD` before being saved or overwritten), but maybe indeed we want to avoid running this detection when the outcome is already known (optimization)?

Reply:
I truly don't know about `Makefile`, actually I did a test, firstly, i compiled by gcc4.8 and used gcc8.2 at second time without `make distclean`, i didn't find i can change `c99` into `c11`.  If you get reach an agreement that i should persist `C11_ATOMIC`, I will do it. @oranagra @yossigo

Reply:
@oranagra It is indirectly controlled by the user (choosing which toolchain to use), and my main concern is some inconsistency (e.g. compilation starts with one compiler/flags, aborted, then continues with another compiler/flags). But you're right, persisting may not make a difference.

I'm okay with leaving it like this, and addressing any issues if/when they show up.